src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -0,0 +1,107 @@
+import csv
+import dataclasses
+import io
+import json
+import re
+from dataclasses import dataclass
+
+import requests
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound
+from databricks.sdk.service.provisioning import Workspace
+
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+from databricks.labs.ucx.hive_metastore.tables import Table
+
+
+class WorkspaceInfoReader:
+    def __init__(self, ws: WorkspaceClient, folder: str):
+        self._ws = ws
+        self._folder = folder
+
+    def _current_workspace_id(self) -> int:
+        headers = self._ws.config.authenticate()
+        headers["User-Agent"] = self._ws.config.user_agent
+        response = requests.get(f"{self._ws.config.host}/api/2.0/preview/scim/v2/Me", headers=headers, timeout=10)
+        return int(response.headers.get("x-databricks-org-id"))
+
+    def _load_workspace_info(self) -> dict[int, Workspace]:
+        try:
+            id_to_workspace = {}
+            workspace_info = self._ws.workspace.download(f"{self._folder}/workspace-info.json")
+            for workspace_metadata in json.loads(workspace_info):
+                workspace = Workspace.from_dict(workspace_metadata)
+                id_to_workspace[workspace.workspace_id] = workspace
+            return id_to_workspace
+        except NotFound:
```

Comment:
Log the exception, as the root cause could have something other than `NotFound`
What if you have permission denied?

Reply:
Need unit and integration tests